File: backend/GastronomicGuide/src/main/java/com/mad/GastronomicGuide/business/RestaurantService.java
package com.mad.GastronomicGuide.business;

import com.mad.GastronomicGuide.entities.Restaurant;

import java.util.List;
import java.util.Optional;

public interface RestaurantService {
    
    List<Restaurant> getAll();
    
    List<Restaurant> getByCity(String city);
    
    Optional<Restaurant> getById(int id);

    List<String> getCities();
}

File: frontend/src/services/restaurantService.js
import axios from "axios"

export default class RestaurantService{

    getRestaurants(){
        return axios.get("http://localhost:8080/api/restaurants/getall")
    }

    getByCity(city){
        return axios.get("http://localhost:8080/api/restaurants/getByCity/?city=" + city)
    }

    getById(id){
        return axios.get("http://localhost:8080/api/restaurants/getById/?id=" + id)
    }

    getCities(){
        return axios.get("http://localhost:8080/api/restaurants/cities")
    }

}